packages/clay-css/src/scss/components/_buttons.scss

```diff
@@ -211,59 +209,99 @@
 
 // Button Variants
 
-@each $color, $value in $btn-palette {
-	.btn-#{$color} {
-		@include clay-button-variant($value);
-	}
-
-	%btn-#{$color}-focus {
-		background-color: map-get($value, focus-bg);
-		border-color: map-get($value, focus-border-color);
-		box-shadow: map-get($value, focus-box-shadow);
-		color: map-get($value, focus-color);
-	}
+@function ccp-color-yiq($color) {
+	//find the text-color using the lightness of hsl colors
+	@return white; //temporary output
 }
 
-// Button Outline Variants
-
-@each $color, $value in $btn-outline-palette {
-	.btn-outline-#{$color} {
-		@include clay-button-variant($value);
-	}
-
-	%btn-outline-#{$color}-focus {
-		background-color: map-get($value, focus-bg);
-		border-color: map-get($value, focus-border-color);
-		box-shadow: map-get($value, focus-box-shadow);
-		color: map-get($value, focus-color);
+$ccp-btn-palette: () !default;
+$ccp-btn-palette: map-deep-merge(
+	(
+		primary: (
+			h: 219,
+			s: 100%,
+			l: 52%,
+		),
+		secondary: (
+			h: 237,
+			s: 8%,
+			l: 46%,
+		),
+		success: (
+			h: 134,
+			s: 52%,
+			l: 32%,
+		),
+		info: (
+			h: 219,
+			s: 58%,
+			l: 43%,
+		),
+		warning: (
+			h: 26,
+			s: 100%,
+			l: 36%,
+		),
+		danger: (
+			h: 0,
+			s: 83%,
+			l: 47%,
+		),
+		light: (
+			h: 225,
+			s: 17%,
+			l: 95%,
+		),
+		dark: (
+			h: 235,
+			s: 13%,
+			l: 18%,
+		),
+	),
+	$ccp-btn-palette
+);
+
+@each $color, $value in $ccp-btn-palette {
+	.btn-#{$color} {
+		--btn-color-h-default: #{map-get($value, h)};
+		--btn-color-s-default: #{map-get($value, s)};
+		--btn-color-l-default: #{map-get($value, l)};
+		--btn-color: hsl(
+			var(--btn-color-h, var(--btn-color-h-default)),
+			var(--btn-color-s, var(--btn-color-s-default)),
+			var(--btn-color-l, var(--btn-color-l-default))
+		);
```

Comment:
the `--btn-color-*-default` are not necessary to define the color but needed to apply variations in the `hover` and `focus` state